File: app/src/main/java/co/medellinandroid/spotifyartists/topTenSongs/TopTenSongsContract.java
package co.medellinandroid.spotifyartists.topTenSongs;

import co.medellinandroid.spotifyartists.model.SpotifyTrack;
import java.util.List;

public interface TopTenSongsContract {

  interface View {
    void showProgress();

    void hideProgress();

    void showError(String message);

    void showSongs(List<SpotifyTrack> tracks);
  }

  interface Presenter {
    void attachView(View view);

    void getTopTenSongs(String artistId);

    void destroy();
  }
}

File: app/src/main/java/co/medellinandroid/spotifyartists/topTenSongs/TopTenSongsContract.kt
package co.medellinandroid.spotifyartists.topTenSongs

import co.medellinandroid.spotifyartists.model.SpotifyTrack

interface TopTenSongsContract {

  interface View {
    fun showProgress()

    fun hideProgress()

    fun showError(message: String)

    fun showSongs(tracks: List<SpotifyTrack>)
  }

  interface Presenter {
    fun attachView(view: View)

    fun getTopTenSongs(artistId: String)

    fun destroy()
  }
}
